themes/default/content/docs/support/troubleshooting.md

```diff
@@ -520,43 +362,140 @@ value defined elsewhere in your application. If the value is known, it can be co
+```bash
+$ pulumi cancel
+...
+The currently running update for 'interruptedstack' has been canceled!
+```
 
-Get in touch with us, either through the [Community Slack](https://slack.pulumi.com) or by emailing our support team: [support@pulumi.com](mailto:support@pulumi.com).
+If `pulumi cancel` fails with `error: [400] Bad Request: the update has already completed`, you can safely ignore
+that error and continue with the next step.
 
-## I can't connect to the Pulumi Service
+You should then export and import your stack. This will clear the stacks's pending operations.
 
-If your network blocks external traffic and you're using the Pulumi Service to manage your state, your security team may need the following details to allow the Pulumi CLI to connect to the Service:
+```bash
+$ pulumi stack export | pulumi stack import
+warning: removing pending operation 'creating' on '...' from snapshot
+Import successful.
+```
 
-* The URL that the Pulumi CLI uses to connect to the Service is `https://api.pulumi.com`. (It does not use `https://app.pulumi.com`, so if you want to view the console, you'll need enable that as well.)
-* All access goes over HTTPS via port 443.
+For every warning that this command prints out, you should verify with your cloud provider whether or not this
+operation was successful. If the operation was successful, and a resource was created, you should delete that
+resource using your cloud provider's console, CLI, or SDK.
+
+Finally, you should run `pulumi refresh` to synchronize your stack's state with the true state of your cloud
+resources:
+
+```bash
+$ pulumi refresh
+Refreshing stack 'interruptedstack'
+Performing changes:
+
+   Type  Name  Status     Info
+
+   info: no changes required:
+         12 resources unchanged
+```
+
+At this point your stack should be valid, up-to-date, and ready to accept future updates.
+
+## Manually Editing Your Deployment {#editing-your-deployment}
+
+Sometimes the only recourse for fixing a stack that is unable to do deployments is to edit the
+deployment directly. We would love to hear about the issues you are experiencing
+that you can't resolve, both so we can assist you in fixing your stack and also to fix the issues in Pulumi
+that made it impossible for you to recover your stack in any other way.
+
+The Pulumi engine uses both your program and your stack's existing state to make decisions about what
+resources to create, read, update, or delete. The most common problem that makes it impossible to
+make changes to your stack is that the stack's existing state has been corrupted in some way. There
+are a variety of ways that a stack's state could be corrupted, but in almost all cases it is possible
+to manually edit the stack's existing state to fix the corruption.
+
+This is an advanced operation and should be an absolute last resort. We recommend you check-in with the [Pulumi Community Slack](https://slack.pulumi.com) first before editing your snapshot.
```

Comment:
```suggestion
This is an advanced operation and should be an absolute last resort. We recommend you check-in with the [Pulumi Community Slack](https://slack.pulumi.com) first before editing your state file.
```

Reply:
@jkodroff i recently learned that snapshot is slightly different than state file, so would like to leave this as is. but i will be adding snapshot to the glossary soon. also, snapshot is mentioned a bunch more times in this file.